GVFS/GVFS.UnitTests.Windows/Windows/Service/ProductUpgradeTimerTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Service;
+using GVFS.UnitTests.Mock.Common;
+using GVFS.UnitTests.Windows.Mock.Upgrader;
+using Moq;
+using Moq.Protected;
+using NUnit.Framework;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GVFS.UnitTests.Windows.Windows.Service
+{
+    public class ProductUpgraderTimerTests
+    {
+        private string response_1 = @"{""Version"":""1.2.3.4""}";
+
+        private interface IHttpMessageHandlerProtectedMembers
+        {
+            Task<HttpResponseMessage> SendAsync(HttpRequestMessage message, CancellationToken token);
+        }
+
+        [TestCase]
+        public void QueriesGitHubForUpgrade()
+        {
+            MockTracer tracer = new MockTracer();
+            Mock<PhysicalFileSystem> fileSystemMock = new Mock<PhysicalFileSystem>();

```

Comment:
I am not opposed to this in cases where we set up expectations exactly (or maybe we can always set up expectations exactly). It might get a little harder with some methods that have more complex signatures with out params (which is a lot of our code).